### PR TITLE
Some features:

### PR DESCRIPTION
**Pull request by antechrestos**
- refactoring of secret management
- upgrade service instance creation for acceptance by passing a dictionnary

### CODE REVIEW

File: src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java [hunk 1]
```diff
@@ -62,6 +62,8 @@ private AutosleepServiceInstance() {
 
     public AutosleepServiceInstance(CreateServiceInstanceRequest request) throws HttpMessageNotReadableException {
         super(request);
+        //only set secret on creation
+        setSecretFromParams(request.getParameters());
```
Comment (arnaudruffin) [inline src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java]: This param is not mandatory on creation. It's mandatory whenever we want to set a protected parameter. Exemple: create service without no_optout (and no secret). Update service with no_optout and secret

Comment (antechrestos) [inline src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java]: Yeah I know: that's why it is not mandatory in the setter. I only set it in a create context (with Create Request). The authorization only compare the one in the request with the one in the object

Comment (arnaudruffin) [inline src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java]: Read my example, it  wouldn't work

Comment (antechrestos) [inline src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java]: Need to discuss if secret may be added on update context

**Pull request merged**